Neos.Neos/Classes/Domain/Import/DoctrineMigrateProcessor.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\Import;
+
+use Neos\ContentRepository\Export\ProcessingContext;
+use Neos\ContentRepository\Export\ProcessorInterface;
+use Neos\Flow\Persistence\Doctrine\Service as DoctrineService;
+
+final readonly class DoctrineMigrateProcessor implements ProcessorInterface
+{
+    public function __construct(
+        private DoctrineService $doctrineService,
+    ) {
+    }
+
+    public function run(ProcessingContext $context): void
+    {
+        $this->doctrineService->executeMigrations();
```

Comment:
doing that in the same process is unsafe when not closing and reopening the connections as a lot of transactions will be nested: see https://github.com/neos/behat/pull/40

Reply:
This is for convenience mostly. Most people will have the migrations applied and it may help some.  I do not see where this would do more harm than good. If you try to import with incomplete migrations failure is guaranteed.

Reply:
Ah i think you misunderstood me: My thesis this will not work as is but needs a $this->connection->close() hotfix to reset the transactions as otherwise the event store doesnt work (see https://github.com/neos/neos-development-collection/pull/5284) ...

Reply:
and now that i think of it, maybe the user doesnt gain anything here as there is not that good output / error handling than if run via `doctrine:migrate`

and now there is a huge overlap to what `flow setup` advises to do and what is done automatically - if it works....

ill think about what `flow setup`'s responsibilities are and what is part of the site setup. https://github.com/neos/neos-setup

But i think setting up migrations is not responsibility (but we can surely check that beforehand and give helpful errors with run `flow setup` first or sth ...)

Reply:
So you suggest to remove the doctrine migrations but keep cr:setup here. To me that would be fine.

Reply:
As written in https://github.com/neos/neos-development-collection/pull/5307#issuecomment-2453048011 actually now that i think of it i would keep the fingers from adjusting the schema hear, and only warn if not done.